src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .map((styles) => styles.trim())
+    .filter((styles) => styles !== '')
+    .reduce((cssStyles, styles) => {
+      const [key, value] = styles.split(':').map((item) => item.trim());
+
+      if (key && value) {
+        cssStyles[key] = value;
+      }
+
+      return cssStyles;
+    }, {});
```

Comment:
The function `convertToObject` is well-implemented using a combination of `split`, `map`, `filter`, and `reduce` which aligns with the checklist of avoiding loops and not modifying existing arrays. However, the variable naming within the `reduce` callback could be improved for better readability. Instead of `cssStyles` and `styles`, consider using names that reflect their roles more clearly, such as `accumulator` or `styleObject` for the `cssStyles` and `styleDeclaration` for `styles`.